lib/screens/play/card/card_pile.dart

```diff
@@ -1,33 +1,43 @@
 import 'package:flutter/material.dart';
+import 'package:treasure_of_the_high_seas/model/game_state.dart';
+import 'package:treasure_of_the_high_seas/screens/play/view_discard_page.dart';
 
 class CardPile extends StatelessWidget {
   final int pileSize;
   final String pileName;
 
-  CardPile(this.pileSize, this.pileName);
+  final GameState state;
+
+  CardPile(this.pileSize, this.pileName, this.state);
 
   @override
   Widget build(BuildContext context) {
-
     final shape = RoundedRectangleBorder(
       side: BorderSide(color: Colors.black, width: 1),
       borderRadius: BorderRadius.circular(10),
     );
 
-    return Center(
-      child:  Card(
-        shape: shape,
-        child: Padding(
-          padding: EdgeInsets.fromLTRB(4, 4, 4, 4),
-          child: Text(
-            pileName + ": " + pileSize.toString(),
-            style: TextStyle(
-              fontSize: 12.0,
-              fontWeight: FontWeight.bold,
-            ),
-          ),
-        )
+    final text = Text(
+      pileName + ": " + pileSize.toString(),
+      style: TextStyle(
+        fontSize: 12.0,
+        fontWeight: FontWeight.bold,
       ),
     );
+
+    return Center(
+        child: Card(
+            shape: shape,
+            child: Padding(
+                padding: EdgeInsets.fromLTRB(4, 4, 4, 4),
+                child: pileName == 'Deck'
+                    ? TextButton(child: text)
+                    : TextButton(
+                        onPressed: pileSize > 0
+                            ? () {
+                                ViewDiscardPage(state);
```

Comment:
This is a nearer miss than the state example, but still doesn't show anything - because the onPressed is constructing a `ViewDiscardPage` but not actually doing anything with it. Instead, we just want to update the state from here - that will update our boolean, and call `notifyListeners` - which will in turn trigger Flutter to automatically rerender our components.

It's probably easiest to explain with a diagram:

![image](https://user-images.githubusercontent.com/5732536/105550904-e267d600-5cf9-11eb-98c5-13661b42702e.png)